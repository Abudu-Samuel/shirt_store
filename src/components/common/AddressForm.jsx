import React from 'react';

export default ({ address, city, postalCode, country, handleChange, shipping_region_id, setRegion, state }) => (
  <div>
    <div>
      <div className="row">
        <div className="col-md-6 mb-3">
          <div className="md-form">
            <i style={{ fontSize: '1.6rem' }} className="fa fa-address-card prefix grey-text" />
            <label className="font-weight-light label">Your adress</label>
            <input type="text" name="address" id="materialFormCardNameEx" value={address || ''} onChange={handleChange} className={state.validation && state.validation.address ? 'error-input-field' : "initial"} />
            {state.validation && (
              <span className="field-error d-flex justify-content-end">
                {state.validation.address}
              </span>
            )}
          </div>
        </div>
        <div className="col-md-6 mb-3">
          <div className="md-form">
            <i style={{ fontSize: '1.6rem' }} className="fa fa-city prefix grey-text" />
            <input type="text" name="city" id="materialFormCardEmailEx" onChange={handleChange} value={city || ''} className={state.validation && state.validation.city ? 'error-input-field' : "initial"} />
            {state.validation && (
              <span className="field-error d-flex justify-content-end">
                {state.validation.city}
              </span>
            )}
            <label className="font-weight-light label">Your city</label>
          </div>
        </div>
        <div className="col-md-6 mb-3">
          <div className="md-form">
            <i style={{ fontSize: '1.6rem' }} className="fa fa-mail-bulk prefix grey-text" />
            <input type="email" name="postalCode" id="materialFormCardEmailEx" value={postalCode || ''} onChange={handleChange} className={state.validation && state.validation.postalCode ? 'error-input-field' : "initial"} />
            {state.validation && (
              <span className="field-error d-flex justify-content-end">
                {state.validation.postalCode}
              </span>
            )}
            <label className="font-weight-light label">Your postal code</label>
          </div>
        </div>
        <div className="col-md-6 mb-3">
          <div className="md-form">
            <i style={{ fontSize: '1.6rem' }} className="fa fa-globe-europe prefix grey-text" />
            <input type="email" name="country" id="materialFormCardEmailEx" value={country || ''} onChange={handleChange} className={state.validation && state.validation.country ? 'error-input-field' : "initial"} />
            {state.validation && (
              <span className="field-error d-flex justify-content-end">
                {state.validation.country}
              </span>
            )}
            <label className="font-weight-light label">Your country</label>
          </div>
        </div>
        <div className="col-md-12 mb-3">
          <div className="md-form">
            <i style={{ fontSize: '1.6rem' }} className="fa fa-address-card prefix grey-text" />
            <label className="font-weight-light label">Your region</label>
            <select name="shippingRegion" value={shipping_region_id} onChange={setRegion} className={state.validation && state.validation.shipping_region_id ? 'error-input-field browser-default custom-select' : "initial browser-default custom-select"}>
              <option value="1" defaultValue>Open this select menu</option>
              <option value="2">US / Canada</option>
              <option value="3">Europe</option>
              <option value="4">Rest of World</option>
            </select>
            {state.validation && (
              <span className="field-error d-flex justify-content-end">
                {state.validation.shipping_region_id}
              </span>
            )}
          </div>
        </div>
      </div>
    </div>
  </div>
)