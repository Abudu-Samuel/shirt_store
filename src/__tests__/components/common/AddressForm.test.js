import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import AddressForm from "../../../components/common/AddressForm";

const props = {
  state: {
    validation: {}
  }
};

describe("<AddressForm />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<AddressForm {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
