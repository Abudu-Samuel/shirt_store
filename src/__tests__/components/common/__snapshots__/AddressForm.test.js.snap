// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressForm /> renders without crashing 1`] = `
<div>
  <div>
    <div
      className="row"
    >
      <div
        className="col-md-6 mb-3"
      >
        <div
          className="md-form"
        >
          <i
            className="fa fa-address-card prefix grey-text"
            style={
              Object {
                "fontSize": "1.6rem",
              }
            }
          />
          <label
            className="font-weight-light label"
          >
            Your adress
          </label>
          <input
            className="initial"
            id="materialFormCardNameEx"
            name="address"
            type="text"
            value=""
          />
          <span
            className="field-error d-flex justify-content-end"
          />
        </div>
      </div>
      <div
        className="col-md-6 mb-3"
      >
        <div
          className="md-form"
        >
          <i
            className="fa fa-city prefix grey-text"
            style={
              Object {
                "fontSize": "1.6rem",
              }
            }
          />
          <input
            className="initial"
            id="materialFormCardEmailEx"
            name="city"
            type="text"
            value=""
          />
          <span
            className="field-error d-flex justify-content-end"
          />
          <label
            className="font-weight-light label"
          >
            Your city
          </label>
        </div>
      </div>
      <div
        className="col-md-6 mb-3"
      >
        <div
          className="md-form"
        >
          <i
            className="fa fa-mail-bulk prefix grey-text"
            style={
              Object {
                "fontSize": "1.6rem",
              }
            }
          />
          <input
            className="initial"
            id="materialFormCardEmailEx"
            name="postalCode"
            type="email"
            value=""
          />
          <span
            className="field-error d-flex justify-content-end"
          />
          <label
            className="font-weight-light label"
          >
            Your postal code
          </label>
        </div>
      </div>
      <div
        className="col-md-6 mb-3"
      >
        <div
          className="md-form"
        >
          <i
            className="fa fa-globe-europe prefix grey-text"
            style={
              Object {
                "fontSize": "1.6rem",
              }
            }
          />
          <input
            className="initial"
            id="materialFormCardEmailEx"
            name="country"
            type="email"
            value=""
          />
          <span
            className="field-error d-flex justify-content-end"
          />
          <label
            className="font-weight-light label"
          >
            Your country
          </label>
        </div>
      </div>
      <div
        className="col-md-12 mb-3"
      >
        <div
          className="md-form"
        >
          <i
            className="fa fa-address-card prefix grey-text"
            style={
              Object {
                "fontSize": "1.6rem",
              }
            }
          />
          <label
            className="font-weight-light label"
          >
            Your region
          </label>
          <select
            className="initial browser-default custom-select"
            name="shippingRegion"
          >
            <option
              defaultValue={true}
              value="1"
            >
              Open this select menu
            </option>
            <option
              value="2"
            >
              US / Canada
            </option>
            <option
              value="3"
            >
              Europe
            </option>
            <option
              value="4"
            >
              Rest of World
            </option>
          </select>
          <span
            className="field-error d-flex justify-content-end"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
